.github/workflows/acceptance.yml

```diff
@@ -265,3 +339,30 @@ jobs:
 
       - name: Handle artifacts
         uses: ./.github/actions/artifacts
+
+  visual-diff:
+    # This guarantees that we will only schedule Visual Snapshots if all
+    # workflows that generate artifacts succeed
+    needs: [acceptance, frontend, chartcuterie]
```

Comment:
This is an improvement over the previous situation where Visual Snapshots would get scheduled even if one of them do not generate artifacts.

This will require engineers to get the dependent workflows to pass before Visual Snapshot check is triggered.

You can see how "check to trigger visual snapshot" does not get triggered:
<img width="816" alt="image" src="https://user-images.githubusercontent.com/44410/159015205-8a7b9e1b-3dc1-46eb-a4d7-59d533053d16.png">

Reply:
The downside here is that you cannot re-run the visual diff job itself, you'll need to re-run this entire test suite. On the flip-side no one knows how to do that. But earlier this week before we removed the slow query, I was restarting theses workflows and they would take 6 minutes to re-run instead of the entirety of the `acceptance.yml` workflow: https://github.com/getsentry/sentry/actions/workflows/visual-diff.yml

Reply:
Good point.

I can think of two solutions to this problem

- 3 workflows -> workflow that simply says "we're done here"
  - This last task would control whether or not all jobs on acceptance are done (a bit of a waste)

The other option is to make our current visual-diff.yml to check whether or not all the necessary checks on the acceptance workflow have completed successfully before triggering Visual Snapshot.

Would you like me to go back to visual-diff.yml and do a follow up? Or would you want to keep what I have?

I'm fine either way. This was an optimization and I can punt it.

Reply:
@billyvg Let me know what way you would prefer me to go about this (In case you want me to give you that ability back).